### PR TITLE
feat: Wso2 template

### PR DESCRIPTION
**Pull request by andrwils**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@
 node_modules
 oclif.manifest.json
 /output
+/lambda/data
```
Comment (mbystedt) [inline .gitignore:10]: Rebase against main branch. Already there. Although, I appear to have misspelt the rule for the keep file. oops. !/labda/data/.keep should be !/lambda/data/.keep

File: config/templates/wso2/input/inputs.conf.njk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{% import "../../macros/input.njk" as input with context %}
+{{- input.renderTail('app.' + typeTag + '.access', logs_path_http_access_log) }}
+
+{{- input.renderMultilineTail('app.' + typeTag + '.generic', logs_path_app, 'multiline-core-app-regex-' + typeTag) }}
```
Comment (mbystedt) [inline config/templates/wso2/input/inputs.conf.njk:4]: Are there any multiline logs? Also, type should probably end in .carbon (not .generic)

File: config/templates/wso2/filter/filters.conf.njk [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    Key_Name message
+    Parser parser-filter-regex-generic-{{typeTag}}
+    Reserve_Data On
+    Preserve_Key On
+
+# App logs (generic)
+#
+[FILTER]
+    Name modify
+    Match app.{{typeTag}}.generic
```
Comment (mbystedt) [inline config/templates/wso2/filter/filters.conf.njk:25]: carbon?

File: config/templates/wso2/filter/filters.conf.njk [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    Rename tomcat_time @timestamp
+    Add event.kind event
+    Add event.category web
+    Add event.dataset application.log
+    Add @metadata.timestampFormat YYYY-MM-DD-HH:mm:ss.SSS{% if shimDisableMultilineKey %}
+    Add log.file.path /unknown.log
+    Add event.sequence 0{% endif %}{% if tmplTomcatTestApplicationIndex %}
```
Comment (mbystedt) [inline config/templates/wso2/filter/filters.conf.njk:32]: May not be necessary to use the shim as carbon doesn't look multiline.

**Review by mbystedt — CHANGES_REQUESTED** [review]

File: config/server/refactor.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "address": "refactor.bcgov",
+  "proxy": "http://forwardproxy.nrs.bcgov:23128",
+  "fluentBitRelease": "1.8.15",
```
Comment (mbystedt) [inline config/server/refactor.json:4]: Hmm... Should be updating this. ;-)

File: config/templates/wso2/filter/filters.conf.njk (hunk 4 above)
Comment (andrwils) [reply config/templates/wso2/filter/filters.conf.njk:33]: I'll confirm if it's multiline.

Comment (sanjbabu) [reply config/templates/wso2/filter/filters.conf.njk:33]: I do see multiline carbon logs on translate
(/sw_ux/wso2/wso2am/repository/logs/wso2carbon.log)

File: config/templates/wso2/filter/filters.conf.njk [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+# App logs (carbon)
+#
+[FILTER]
+    Name modify
+    Match app.{{typeTag}}.carbon
+    Rename tomcat_time @timestamp
+    Rename tid event.provider
+    Add event.kind event
+    Add event.category web
+    Add event.dataset application.log
```
Comment (andrwils) [inline config/templates/wso2/filter/filters.conf.njk:30]: Do we need the dataset to be carbon.log for the fingerprint?

Comment (GraceRuan) [reply config/templates/wso2/filter/filters.conf.njk:30]: Matthew mentioned it using generic one, no new index with new fields needed in fingerprint. @mbystedt please correct me if I misunderstood it.

Comment (mbystedt) [reply config/templates/wso2/filter/filters.conf.njk:30]: https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-dataset
https://docs.wso2.com/display/ADMIN44x/Monitoring+Logs#MonitoringLogs-log_typesLogtypesinWSO2products

WSO2 refers to them as carbon. I would say it should be wso2.carbon or application.carbon.

Same index as they are generic application logs.

**Review by mbystedt — APPROVED** [review]

**Pull request merged**